doc/moveit_task_constructor/moveit_task_constructor_tutorial.rst

```diff
@@ -67,22 +67,23 @@ Stages can be arranged in any arbitrary order and hierarchy only limited by the
 The order in which stages can be arranged is restricted by the direction in which results are passed.
 There are three possible stages relating to the result flow: generator, propagator, and connector stages:
 
-**Generators** compute their results independent of their neighbor stages and pass it in both directions, backwards and forwards.
+**Generators** compute their results independently of their neighbor stages and pass them in both directions, backwards and forwards.
 An example is an IK sampler for geometric poses where approaching and departing motions (neighbor stages) depend on the solution.
 
-**Propagators** receive the result of one neighbor stage, solve a subproblem and then propagate their result on to the neighbor on the opposite site.
-Depending on the implementation propagating stages can pass solutions forward, backward or in both directions separately.
+**Propagators** receive the result of one neighbor stage, solve a subproblem and then propagate their result to the neighbor on the opposite site.
+Depending on the implementation, propagating stages can pass solutions forward, backward or in both directions separately.
 An example is a stage that computes a Cartesian path based on either a start or a goal state.
 
-**Connectors** are don’t propagate any results, but rather attempt to bridge the gap between the resulting states of both neighbors.
+**Connectors** do not propagate any results, but rather attempt to bridge the gap between the resulting states of both neighbors.
 An example is the computation of a free-motion plan from one given state to another.
 
 Additional to the order types, there are different hierarchy types allowing to encapsulate subordinate stages.
 Stages without subordinate stages are called **primitive stages**, higher-level stages are called **container stages**.
 There are three container types:
 
 **Wrappers** encapsulate a single subordinate stage and modify or filter the results.
-An example is a filter stage that only accepts solutions of an IK solver that satisfy a certain constraint.
+A filter stage that only accepts solutions of its child stage which satisfy a certain constraint can be realized as a wrapper.
```

Comment:
```suggestion
For example, a filter stage that only accepts solutions of its child stage that satisfy a certain constraint can be realized as a wrapper.
```

Reply:
This is what [Merriam Webster](https://www.merriam-webster.com/words-at-play/when-to-use-that-and-which) has to say on the matter:

> but which seems to have a fair degree of flexibility to it, and can perform very nicely in restrictive clauses, nonrestrictive clauses, and in annoying people who feel that it should never be found outside of a nonrestrictive clause

Do you feel annoyed? :grin: 

I do not have hard feelings on this. Feel free to push this before merging if you prefer your version.

Reply:
Interesting. I learned to use _that_ for _restrictive_ and _which_ for _non-restrictive_ clauses.
However, my main point was adding "For example, " :wink: